src/subdomains/supporting/payment/services/transaction-request.service.ts

```diff
@@ -166,7 +240,25 @@ export class TransactionRequestService {
   }
 
   async complete(id: number): Promise<void> {
-    await this.transactionRequestRepo.update(id, { isComplete: true });
+    await this.transactionRequestRepo.update(id, { isComplete: true, status: TransactionRequestStatus.COMPLETED });
+  }
+
+  async extendTransactionRequest(txRequest: TransactionRequest): Promise<TransactionRequestExtended> {
+    const sourceAssetEntity =
+      txRequest.type === TransactionRequestType.BUY
+        ? await this.fiatService.getFiat(txRequest.sourceId)
+        : await this.assetService.getAssetById(txRequest.sourceId);
+
+    const targetAssetEntity =
+      txRequest.type === TransactionRequestType.SELL
+        ? await this.fiatService.getFiat(txRequest.targetId)
+        : await this.assetService.getAssetById(txRequest.targetId);
+
+    return Object.assign(txRequest, { sourceAssetEntity, targetAssetEntity });
+  }
```

Comment:
Can we use this in other places in our code?

Reply:
Hm I think atm not or do you have a example? I cannot find a place atm in our code where we could use it too